Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -407,51 +437,62 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
                 context.writeAndFlush(self.wrapOutboundOut(self.encoder.flushBuffer()), promise: nil)
             }
             context.close(mode: .all, promise: promise)
+            if let promise {
+                promise.futureResult.cascade(to: runPromise)
+            } else {
+                runPromise?.succeed(())
+            }
         case .succeedPreparedStatementCreation(let promise, with: let rowDescription):
             promise.succeed(rowDescription)
+            runPromise?.succeed(())
         case .failPreparedStatementCreation(let promise, with: let error, let cleanupContext):
             promise.fail(error)
+            runPromise?.fail(error)
             if let cleanupContext = cleanupContext {
                 self.closeConnectionAndCleanup(cleanupContext, context: context)
             }
         case .sendCloseSync(let sendClose):
-            self.sendCloseAndSyncMessage(sendClose, context: context)
+            self.sendCloseAndSyncMessage(sendClose, context: context, promise: runPromise)
         case .succeedClose(let closeContext):
-            closeContext.promise.succeed(Void())
+            closeContext.promise.succeed(())
+            runPromise?.succeed(())
         case .failClose(let closeContext, with: let error, let cleanupContext):
             closeContext.promise.fail(error)
+            runPromise?.fail(error)
             if let cleanupContext = cleanupContext {
                 self.closeConnectionAndCleanup(cleanupContext, context: context)
             }
         case .forwardNotificationToListeners(let notification):
             self.forwardNotificationToListeners(notification, context: context)
+            runPromise?.succeed(())
```

Comment:
This also should be good to just succeed the promise here?
Should I handle it somehow else?
Should I pass the promise and succeed it in the function?